docs/adr/adr-013-blocksync-overhaul-part-2.md

```diff
@@ -0,0 +1,530 @@
+          ║ ALT  │  STATUS:OK            │             ║
+          ╟──────┘     │                 │             ║
+          ║            │     CAR File    │             ║
+          ║            │ <────────────────             ║
+          ╚════════════╪═════════════════╪═════════════╝
+                       │                 │              
+                       │   Stream Close  │              
+                       │ <────────────────              
+                    ┌──┴───┐          ┌──┴───┐          
+                    │Client│          │Server│          
+                    └──────┘          └──────┘          
+
+```
+
+### Misbehavior
+
+* Server sends ODS CAR file that fails EDS reconstruction.
+* Client terminates stream after receiving the EDSResponse.
+
+### ShrEx/ND Protocol
+
+ShrEx/ND is a pull-based protocol with client-server model, where LNs are clients and BNs/FNs are servers.
+The protocol has only one request-response interaction to get all the data/shares addressed by namespace
+
+The protocol is designed over plain libp2p streams with `/shrex/nd/0.0.1` as the protocol ID.
+
+#### Protobuf Schema
+
+##### Request
+
+```protobuf
+syntax = "proto3";
+
+message NDRequest {
+  bytes hash = 1; // identifies the EDS to get ND from
+  bytes namespace = 2; // namespace of ND
+}
+```
+
+##### Response
+
+```protobuf
+syntax = "proto3";
+
+enum Status {
+  INVALID = 0;
+  OK = 100; // data found
+  NOT_FOUND = 200; // data not found
+  REFUSED = 201; // request refused
+}
+
+message Proof {
+  int64 start = 1;
+  int64 end = 2;
+  repeated bytes Nodes = 3;
+}
+
+message Row {
+  repeated bytes shares = 3;
+  Proof proof = 4;
+}
+
+message NDResponse {
+  Status status = 1;
+  repeated Row rows = 2;
+}
+
+```
+
+#### Streaming
+
+The v0 version of the protocol sends ND and supporting NMT proofs as a single message. The libp2p has a max message size
+of 1MB, meaning that the `NDResponse` message is limited with 1MB.
+
+The following versions will introduce streaming by either splitting `NDResponse` into multiple messages or by streaming
+data and proofs in real time, while they are read from the disk.
+
+#### Flow
+
+     ┌──────┐          ┌──────┐
+     │Client│          │Server│
+     └──┬───┘          └──┬───┘
+        │   Open Stream   │    
+        │ ────────────────>    
+        │                 │    
+        │    NDRequest    │    
+        │ ────────────────>    
+        │                 │    
+        │    NDResponse   │    
+        │ <────────────────    
+        │                 │    
+        │   Stream Close  │    
+        │ <────────────────    
+     ┌──┴───┐          ┌──┴───┐
+     │Client│          │Server│
+     └──────┘          └──────┘
+
+
+### ShrEx/Sub Protocol
+
+`ShrEx/Sub` is push-based notification protocol with PubSub model, where LNs are subscribers, BNs are publishers and
+FNs are both.
+
+The protocol is based on libp2p's `FloodSub`(TODO Link) with `/ShrEx/sub/0.0.1` as topic ID.
```

Comment:
```suggestion
The protocol is based on libp2p's `FloodSub`(TODO Link) with `eds-sub` as topic ID.
```
can we be consistent with other pubsub topics?

Reply:
Not necessary, but good call

Reply:
Made it `eds-sub`, but I kept versioning